src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -11,20 +11,20 @@
 	 * so they can be adjusted on the fly. 
 	 * Or we can still include it in the JAR but as a separate file
 	 * so we can change it through the development process.  */
-	private static final int QUEUE_SIZE_LIMIT = 10;
+	private static final int STACK_SIZE_LIMIT = 10;
 	
-	private Stack<SudokuAction> sudokuActionQueue;
-	private Stack<SudokuAction> sudokuActionQueueForUndo;
+	private Stack<SudokuAction> sudokuActionStack;
+	private Stack<SudokuAction> sudokuActionStackForUndo;
 
 	public ActionManager() {
-		this.sudokuActionQueue = new Stack<SudokuAction>();
-		this.sudokuActionQueue.setSize(QUEUE_SIZE_LIMIT);		
-		this.sudokuActionQueueForUndo = new Stack<SudokuAction>();
-		this.sudokuActionQueueForUndo.setSize(QUEUE_SIZE_LIMIT);
+		this.sudokuActionStack = new Stack<SudokuAction>();
+		this.sudokuActionStack.setSize(STACK_SIZE_LIMIT);		

```

Comment:
The java standard recommends using ArrayDeque instead of Stack. Then you don't need to worry about the initial size (because resizing is fast), and isEmpty will work for you.